File: Tutorials/05_MotorErrorHandling/main.py
from pyorcasdk import Actuator
import pyorcasdk.orca_registers as orca_reg
import sys

motor = Actuator( "MyMotorName" )

serial_port = int(input("Please input the serial port of your connected motor. "))

serial_port_error = motor.open_serial_port(serial_port)

motor_errors_result = motor.get_errors()

if motor_errors_result.error:
	print("Failed to read active errors: " + motor_errors_result.error.what())
	sys.exit(1)

print("Active motor errors: " + str(motor_errors_result.value))

if (motor_errors_result.value & orca_reg.ERROR_0_VOLTAGE_INVALID_Mask):
    print("There is definitely an invalid supply voltage error!")

File: Tutorials/05_MotorErrorHandling/main.cpp
#include <iostream>
#include "actuator.h"
#include "TutorialHelpers.h"

using namespace orcaSDK;

int main() {
	Actuator motor{ "MyMotorName" };

	int serial_port = obtain_serial_port_number();

	motor.open_serial_port(serial_port);

	OrcaResult<uint16_t> motor_errors_result = motor.get_errors();

	if (motor_errors_result.error)
	{
		std::cout << "Failed to read active errors: " << motor_errors_result.error.what() << "\n";
		return 1;
	}

	std::cout << "Active motor errors: " << motor_errors_result.value << "\n";

	if (motor_errors_result.value & ORCAReg::ERROR_0_Values::VOLTAGE_INVALID_Mask) std::cout << "There is definitely an invalid supply voltage error!" << "\n";

	return 0;
}